server/controllers/chartsController/helpers.test.js

```diff
@@ -739,4 +836,168 @@ describe('chartsController - helpers', () => {
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              socioDemographicsCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+        inclusionCriteria: [
+          {
+            $match: {
+              assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+
+    it('returns an object only for the requested filters', () => {
+      expect(
+        helpers.buildFacetForFilters({
+          isSexAtBirthFilterActive: true,
+          isInclusionCriteriaFilterActive: undefined,
+          userAccess,
+        })
+      ).toEqual({
+        socioDemographics: [
+          {
+            $match: {
+              assessment: SOCIODEMOGRAPHICS_FORM,
+              study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+            },
+          },
+          {
+            $project: {
+              ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+              socioDemographicsCollection: MONGO_COLLECTION_STRING,
+            },
+          },
+        ],
+      })
+    })
+  })
+
+  describe(helpers.intersectSubjectsFromFilters, () => {
+    it('adds subjects to the resultsFromFiltersMap', () => {
+      const requestedFilters = {
+        socioDemographics: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            socioDemographicsCollection:
+              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
+          },
+        ],
+        inclusionCriteria: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            inclusionCriteriaCollection:
+              '062419c48a8b6634e696b9dc0674fb87c34aea44751d1eba107feb0fc7325d71',
+          },
+          {
+            study: 'CA',
+            subject: 'CA00063',
+            inclusionCriteriaCollection: 'CA00063filter',
+          },
+          {
+            study: 'LA',
+            subject: 'LA00028',
+            inclusionCriteriaCollection: 'la00028filter',
+          },
+          {
+            study: 'YA',
+            subject: 'YA00015',
+            inclusionCriteriaCollection: 'YA00015Filter',
+          },
+        ],
+      }
+
+      expect(
+        Object.keys(helpers.intersectSubjectsFromFilters(requestedFilters))
+          .length
+      ).toEqual(1)
+    })
+
+    it('returns a map with subject data merged', () => {
+      const requestedFilters = {
+        socioDemographics: [
+          {
+            study: 'YA',
+            subject: 'YA01508',
+            socioDemographicsCollection:
+              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
+          },
```

Comment:
I think if we add a subject here that isn't in the other filters we will expose a bug in the intersection logic.

```suggestion
          {
            study: 'YA',
            subject: 'YA01508',
            socioDemographicsCollection:
              '99fc2c55c6718fcd8eadf9244a24d2f68aae77c6fc591d5e2a8e67bab906446f',
          },
          {
            study: 'FOO',
            subject: 'FOO01508',
            socioDemographicsCollection: 'foo-collection',
          },
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/cbd951fa9f62d42a3ae53518c2f1c4b48f02635b